### DISCUSSION TITLE
Can't find node %PhantomCamera2D

### DISCUSSION THREAD

**Discussion by ChildishGiant** [ramokz/phantom-camera#249] [Question / Help Wanted] [answered]
I'm very new to Godot still so this is probably a stupid mistake but I can't seem to reference the camera from these instantiated nodes 

![image](https://github.com/ramokz/phantom-camera/assets/13716824/0d7f09d3-a92b-4bfb-a6f2-6c64303bbe6e)
![image](https://github.com/ramokz/phantom-camera/assets/13716824/861276e2-68dc-4066-8a05-9754168910f3)

**Comment by ramokz** [comment]
Is the script you're showing attached to all the `0_0-0`, `0_1-0` etc. nodes?

**Reply by ChildishGiant** [reply]
Yeah, each of those has (or should have at least) tile.gd attached to it.

**Reply by ramokz** [reply, marked as answer]
Based on the error output, it says that it's looking for `%PhantomCamera2D` as a child inside the tile nodes. But because the `pcam` node is a sibling node, it should just be a case of changing the relative path. This might not be the most concise way of doing it, but you can always assign it in the _ready() method by doing, for example.
```gdscript

var pcam: PhantomCamera2D

func _ready():
  pcam = get_parent().get_node("%PhantomCamera2D")
```

In the odd chance that there are some instantiation race conditions, you can also verify that the node is ready by using an await signal like so:
```gdscript
func _ready():
  var pcam_temp: PhantomCamera2D = get_parent().get_node("%PhantomCamera2D")

  if not pcam_temp.is_node_ready(): await pcam_temp.ready
  pcam = pcam_temp
```
